modules/index-service/src/main/java/org/opencastproject/index/service/impl/index/event/RetractionListener.java

```diff
@@ -54,22 +55,32 @@ public void stateChanged(WorkflowInstance workflow) {
     if (!retractions.containsKey(workflow.getId())) {
       return;
     }
-    final Retraction retraction = retractions.get(workflow.getId());
-    SecurityUtil.runAs(securityService, retraction.getOrganization(), retraction.getUser(), () -> {
-      final String mpId = workflow.getMediaPackage().getIdentifier().compact();
-      try {
-        final boolean result = indexService.removeEvent(mpId);
-        if (!result) {
-          logger.warn("Could not delete retracted media package {}. removeEvent returned false.", mpId);
+    MediaPackage mediaPackage = workflow.getMediaPackage();
+    if (mediaPackage == null) {
+      logger.warn("The retract workflow \"{}\" (id: {}, created by: {}) does not have a media package.",
+              workflow.getTitle(), workflow.getId(), workflow.getCreatorName());
+    } else if (mediaPackage.getPublications() != null && mediaPackage.getPublications().length > 0) {
+      logger.warn("The retract workflow \"{}\" (id: {}, created by: {}, media package {}) has leave some publications.",
```

Comment:
`has leave some publications` -> `has some non-retracted publications, refusing to orphan them.`

Reply:
Hi @gregorydlogan, I'm not shure what you are mean. We can't refuse the deletion from the UI anymore because this will conflict with the behavior of one-click-delete. Moving this part to the asset manager isn't right i.m.o. because you will forbid to delete events with existing publications in general . But I agree with you, it would be right to put the event in an error state (without failed workflows?!) or setting an error comment on it.